core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -259,6 +259,40 @@ public PasswordAuthentication getPasswordAuthentication() {
         return con;
     }
 
+    /**
+     * This method should be used wherever {@link URL#openConnection()} to internet URLs is invoked directly and a redirection is possible.
```

Comment:
Does not actually explain that this allows cross-protocol redirects.

Especially the downgrade from HTTPS to HTTP can be problematic, so it's not something that should just be recommended this freely.

Reply:
Changed.